README.md

```diff
@@ -1,108 +1,207 @@
+- [Housekeeping] System will delete old videos when remaining storage space falls below 20%
 
 ## Overview
+
 As of late 2018 Tesla released V9 which among a number of improvements included dash cam functionality. This works by placing a suitably sized USB drive in one of the available USB ports at the front of the vehicle (Model S).
 
 One drawback of this system, not uncommon in dash cams, there's no easy way to push this video to the 'cloud' - nor any capability to view in near real-time. This project aims to make this possible.
 
 Using a couple of tricks I've learned through tinkering with various single board computers, it is possible emulate a USB drive on the fly. In essence we are going emulate a USB drive, and periodically store the data on the SDHC. Once we have the video we can do what ever we'd like - maybe live stream, upload to your favourite cloud provider or simply backup the files when you return home.
 
 ## Hardware Requirements
+
 1. 2017 (AP 2.5) or beyond Tesla
 2. Raspberry Pi Zero W (only this model is supported)
 3. A wireless access point within reasonable distance of the Pi (mobile phone, home router etc)
-4. A sufficiently large SDHC card with the fastest write speeds you can find, at least 16Gig, ideally the largest you can buy. 
+4. A sufficiently large SDHC card with the fastest write speeds you can find, at least 16Gig, ideally the largest you can buy.
 5. High quality short USB A to USB Micro cable - Anker is quite decent
 6. Optional, a case to house the Raspberry Pi - anything with ventilation would be fine
 
 ## Software Requirements
+
 1. 2018-11-13-raspbian-stretch-lite or later
 2. Etcher to write the disk image to the SDHC card (dd, win32diskimager etc etc will also work)
 3. daemontools package
 4. NodeJS 10.x for Arm v6
 5. OTG Mode enabled in the boot configuration
 6. [Dropbox uploader](https://github.com/andreafabrizi/Dropbox-Uploader)
 
+### Optional Docker Instructions
+
+#### Software install instructions
+
+1. Etcher -> Raspbian
+1. Install Docker (getdocker curl)
+
+   - note about locking the version due to the [bug](https://github.com/moby/moby/issues/38175#issuecomment-463164203).
+     `/etc/apt/preferences.d/docker-ce`
+
+   ```
+   Package: docker-ce
+   Pin: version 18.06.*
+   Pin-Priority: 1000
+   ```
+
+   Install docker ce:
+
+   ```
+   $ curl -fsSL get.docker.com -o get-docker.sh && sh get-docker.sh
+   ```
+
+1. Setup OTG Mode
+
+#### Running
+
+The container uses SSH to talk back to the host system and run commands. Because of this there is a one time setup required to get credentials authorized.
+
+1. Generate SSH Key
+   ```
+   $ docker run \
+   --rm \
+   -it \
+   -v /home/pi/etc/ssh:/root/.ssh \
+   --entrypoint "ssh-keygen" \
+   wurmr/dashcam-monitor
+   ```
+1. Copy SSH Key
+   ```
+   $ docker run \
+   --rm \
+   -it \
+   -v /home/pi/etc/ssh:/root/.ssh \
+   --add-host dockerhost:172.17.0.1 \
```

Comment:
We should parameterise the IP addresses and users

Reply:
Yes, everything should be configurable. Added a todo to the main bit.

Also the `--add-host` param will be going away, it is in there temporarily but we do not need it as soon as I properly elevate all config.